### ISSUE TITLE
Add meaningful error messages for failed Django login on login page

### ISSUE THREAD

**Issue by axoplasm** [mercycorps/TolaActivity#1342]
Currently, failing to enter a valid u/p combination on the Django login returns me silently to the login page. We need to add:

- [ ] ~link to reset password or other help]~ _(no longer addressing in this ticket)_
- [x] error message(s)

**Comment by jennyhavah** [comment]
I replicated the experience, and think it would be unfortunate to leave it this way for release. This is what I see if I login to the old Django admin with a bad username/password:

![image](https://user-images.githubusercontent.com/33670923/55918481-2848be00-5ba8-11e9-8922-d793e63622d2.png

Because Paul has so many tickets in flight right now, I'd like to ask @camgnostic or @KennyMonster if you could take a peek at this and let me know whether it would take more than a few hours to do something along the lines of what the old admin does?

**Comment by KennyMonster** [comment]
@jennyhavah An error message on failed login should be easy enough.

Password reset workflow is a whole thing though, and wanted to verify that the stock Django workflow is desired and worth the time investment. From memory, I think the workflow is:

* user clicks link that they forgot their password
* they are taken to a form where they type in their email
* user is sent to a page saying an email is sent (or something went wrong)
* an email is sent to the user containing a link to a page where they enter in a new password
* the user clicks the link, types in their PW twice, and submits
* a success page is shown saying the reset succeeded

This requires 4 new pages (request reset form + email sent success page + actual reset page + reset success page) and some verbiage for the email sent to the user. Overall it's doable but a bit involved for our tight schedule.

**Comment by jennyhavah** [comment]
I would be fine with just the error message for the Papaya release; we can pull the password reset flow into another ticket and put it in Poblano.

@axoplasm does that work for you?

**Comment by RickieT** [comment]
**QA - PASSED**
Outside of adding a reset workflow, it looks like were now covering the error messaging for failed login credentials, pending @axoplasm response, I think this passes the criteria for this ticket. 

![image](https://user-images.githubusercontent.com/23220180/56056265-898ba100-5d10-11e9-87cc-872c6759c9b5.png)

![image](https://user-images.githubusercontent.com/23220180/56056305-a0ca8e80-5d10-11e9-9029-1bce5951f603.png)

**Comment by RickieT** [comment]
@jennyhavah I was thinking alternativley, we could add the same message to the django login page "Need help logging in? Contact your TolaData administrator or email the TolaData team."

**Comment by jennyhavah** [comment]
@RickieT I like the idea of adding some instructional language to the old Django admin, people may have it bookmarked or miss the announcements about the new admin. They'll go to the old one and wonder why they can't log in anymore. 

@KennyMonster Would it be easy or hard to add a message _only in the old Django admin_, underneath the sign-in box? This is in addition to the existing error message, not instead of. Here's what I'd say:

Are you a TolaActivity admin having trouble logging in? As part of our Partner Access release, we incorporated new and improved administration tools into TolaActivity itself. Please visit TolaActivity and look for a section called Admin.

**Comment by jennyhavah** [comment]
@RickieT Moved this to Doing to give Ken a chance to review and respond to the message above.

**Comment by KennyMonster** [comment]
@jennyhavah Would this work?

![Screen Shot 2019-04-12 at 1 45 06 PM](https://user-images.githubusercontent.com/1278224/56065270-36721800-5d29-11e9-9daa-8dc1b73203c1.png)

**Comment by jennyhavah** [comment]
Looks good to me, but I'd like to just run this by @hannalcamp because it kind of falls in the realm of communications/training, so want to get her approval on the language, etc.

@RickieT Let's leave this in Doing until she gets a chance to review -- she's on her way back home and will be checking in later today or Monday.

**Comment by hannalcamp** [comment]
Messaged to @KennyMonster separately that at first checking this I missed the grey text at bottom at first, and I'm assuming a lot of Admins will be as lazy as I was and not read all the way through to the end. Just hoping we can make that text more noticeable before closing this out, otherwise looks good.

**Comment by KennyMonster** [comment]
I changed the grey text to be conditional on the error message, such that it only shows up if a login error is also on display.

Initial login form:

![Screen Shot 2019-04-15 at 9 57 29 AM](https://user-images.githubusercontent.com/1278224/56150976-01eaa000-5f65-11e9-9e23-0bf23aa92ae4.png)

If there is an error logging in:

![Screen Shot 2019-04-15 at 9 57 18 AM](https://user-images.githubusercontent.com/1278224/56150988-08791780-5f65-11e9-85c9-9ac3dbcec6d0.png)

**Comment by hannalcamp** [comment]
Looks good to me.

**Comment by jennyhavah** [comment]
From a visual standpoint, I think it's a little odd to have some of the message inside the box and some outside, with the form getting pushed down to the bottom.

I would recommend moving the gray text back below the Login button, but keeping it conditional, so that it is still much more noticeable than if we had it there from the start.

**Comment by RickieT** [comment]
**QA - PASSED** 
* Looks like we're all set here: 
![Screen Shot 2019-04-15 at 12 53 26 PM](https://user-images.githubusercontent.com/23220180/56161059-80533c00-5f7d-11e9-9723-079033914602.png)

**Comment by hannalcamp** [comment]
If it remains conditional on error at login, I think that's fine.

**Comment by axoplasm** [comment]
I won't undo any work here but I'm pulling the ticket back into This Release as its related to the auth epic #1443

**Comment by RickieT** [comment]
**QA - PASSED (Again)**
* This is continuing to function as intended after #1443 changes:
![Screen Shot 2019-04-23 at 11 26 06 AM](https://user-images.githubusercontent.com/23220180/56606257-a9e21800-65ba-11e9-9afb-8836444f7f51.png)
